### ISSUE TITLE
mktemp: cannot create temp file /tmp/alias_completion-10624XXX.tmp: Invalid argument

### ISSUE THREAD

**Issue by juboba** [Bash-it/bash-it#1158]
I don't know if this is due to my version of `mktemp`, but the template provided seems not to be valid.
I fixed this simply by changing the template.
In *plugins/available/alias-completion.plugin.bash:
```
-    local tmp_file; tmp_file="$(mktemp -t "${namespace}-${RANDOM}XXX.tmp")" || return 1
+    local tmp_file; tmp_file="$(mktemp -t "${namespace}-${RANDOM}XXXXXX")" || return 1
```

* mktemp version debianutils-2.7
* bash-it branch: master

**Comment by nwinkler** [comment]
Is there an error when using the existing version? What happens in that case?

(Trying to understand what "seems not to be valid" means)

**Comment by juboba** [comment]
Oh, sorry. I meant that the template provided to `mktemp` is invalid.
Calling mktemp with too few X's gives error. I guess that it requires a minimum of X's to ensure uniqueness.

**Comment by nwinkler** [comment]
I tried this on macOS and did not run into any issues. On the other hand, your updated statement worked fine as well - I don't see any issues with changing that.

Can you please create a Pull Request for the change?

**Comment by juboba** [comment]
Sure, I'll create the pull request. It would be nice to know why this happens.
I know OSX uses a different set of tools so maybe that's the issue (`find`, for instance, works slightly different in OSX systems than in typical Linux systems).

**Comment by nwinkler** [comment]
Good question... I'm pretty sure I tested this on a Linux VM, too, and did not find any issues with that command - but I know that there are some subtle (and some not so subtle)  differences between the Gnu and BSD tools.

**Comment by rico-chet** [comment]
TL;DR: close this

As of today, `mktemp` is provided by the `coreutils` package in _debian sid_:

```
$ apt-file search --regexp '/mktemp$'
bash-completion: /usr/share/bash-completion/completions/mktemp
coreutils: /bin/mktemp
```
And it happily accepts the original template, which is now replaced as suggested by the submitter by the way. So I cannot reproduce the original issue and the proposed fix is merged anyway, so this can be closed.

BTW the only thing I can think of causing this is that the submitter accidentally hooked up `busybox`'s version of `mktemp` which doesn't support suffixes as it seems:
```
$ busybox mktemp -t "n-123456XXX.tmp"
mktemp: Invalid argument
```

**Comment by nwinkler** [comment]
Thanks for confirming - closing this now!
